In [ ]:
%pip install statsmodels
from statsmodels.stats.power import TTestIndPower


In [9]:
import numpy as np


# Week 7 Interactive
### Diya, Sharmil, Tiffany
#### GratiText 

### 1. What are your null and alternative hypotheses?

Research Question: Do people who receive a daily gratitude text report 2x higher happiness levels than people who don't over 4 weeks?

Null Hypothesis: There is no difference in happiness scale levels between the group receiving the text and the control group after 4 weeks.

Alternative Hypothesis: The group receiving the text has a higher happiness scale level than the control group after 4 weeks.

### 2. What statistical test will you perform?

We will be using the Mann-Whitney U statistical test. We'll be using this test because it compares whether 2 independent groups differ in their ordinal outcomes without assuming normality. 

Our 2 independent groups are the group receiving the daily gratitude text and the control group. Our outcome variable is the happiness scale. This is an ordinal variable that has a meaningful ranking/order. And we want to test if one group tends to have higher values than the other by comparing median differences.

For Mann-Whitney U statistical test, we will be checking the assumptions:
- Verify each participant's score must be independent from another's score.
- Verify the data values are ordinal.
- Verify the samples are selected randomly.
- Verify that the groups have similar-shaped distributions.

### 3. Explore literature to find effect size
For effect size, we'll be using the Odds Ratio for finding the effect size. 
Small effect: OR ≈ 1.68
Medium effect: OR ≈ 3.47
Large effect: OR ≈ 6.71
Found from - https://doi.org/10.1080/03610911003650383

### 4. Perform power analysis with effect size

In [11]:
OR = 3.47
alpha = 0.05
power = 0.80

effect_size = np.log(OR) * np.sqrt(3) / np.pi   # calculates Cohen's d value
print(effect_size)
analysis = TTestIndPower()
sample_size = analysis.solve_power(effect_size=effect_size, alpha=alpha, power=power, alternative='larger')
print("Sample size", sample_size)

0.6859383780212357
Sample size 26.98286317157412


Our medium effect-size by Odds Ratio is 3.47 this is equivalent to Cohen's d of ~0.686. We need to calculate Cohen's d because this TTestIndPower only takes Cohen's d as the effect size. 

Based on this power analysis, we found that our medium sample size is 27 people per group (total of 54 participants) to detect a medium effect size with 80% power.

### 5. Summary
We have provided rationale for each decision above.

Hypothesis : One-sided, two-group comparison of median

Statistical test : Mann-Whitney U statistical test

Outcome measure : Odds Ratio

Effect size estimate : Odds Ratio = 3.47 (medium effect)

Power analysis result : 27 participants per group (total N = 54)